### PR TITLE
CentOS 8 support

### PR DESCRIPTION
**Pull request by TheREK3R**
some changes def need to be made package wise, but I am not sure what needs to be added/removed

### CODE REVIEW

File: Berksfile [hunk 1]
```diff
@@ -15,5 +14,5 @@ cookbook 'osl-nrpe', git: 'git@github.com:osuosl-cookbooks/osl-nrpe'
 cookbook 'osl-postfix', git: 'git@github.com:osuosl-cookbooks/osl-postfix'
 cookbook 'osl-postgresql', git: 'git@github.com:osuosl-cookbooks/osl-postgresql'
 cookbook 'osl-selinux', git: 'git@github.com:osuosl-cookbooks/osl-selinux'
-
+cookbook 'percona', git: 'git@github.com:sous-chefs/percona.git'
```
Comment (ramereth) [inline Berksfile:17]: This isn't needed since we'll pull from the supermarket, please remove.

File: libraries/helpers.rb [hunk 2]
```diff
@@ -1,9 +1,18 @@
-def sudo_commands(*args)
-  args.reduce([]) { |acc, elem| acc + systemctl_commands(elem) }
-end
+module OslApp
+  module Cookbook
+    module Helpers
+      def sudo_commands(*args)
```
Comment (ramereth) [inline libraries/helpers.rb:4]: It might be a good idea to namespace these methods with `osl_`

File: recipes/default.rb [hunk 3]
```diff
@@ -31,19 +34,25 @@
   'a9d9e6308931fa2a2b0cada070516d45b76d752430c31c9198933c78f8d54b17'
 
 # rvm package depends
-%w(sqlite-devel libyaml-devel readline-devel zlib-devel libffi-devel
```
Comment (ramereth) [inline recipes/default.rb:34]: Why was this changed?

File: recipes/default.rb [hunk 4]
```diff
@@ -31,19 +34,25 @@
...
   package p
 end
 
 # geo-django depends
-%w(gdal-python geos-python proj proj-nad postgresql-devel
-   freetype-devel libpng-devel libjpeg-turbo-devel postgis
-   python-psycopg2).each do |p|
+%w(geos-devel proj
+   freetype-devel libpng-devel libjpeg-turbo-devel
+  ).each do |p|
```
Comment (ramereth) [inline recipes/default.rb:46]: Please make this one line.

File: recipes/default.rb [hunk 5]
```diff
@@ -31,19 +34,25 @@
...
   package p
 end
 
+if node['platform_version'].to_i < 8
+  package 'python-psycopg2'
+else
+  package 'python2-psycopg2'
+end
```
Comment (ramereth) [inline recipes/default.rb:54]: This is a better way to do this IMHO.
```suggestion
package 'python-psycopg2' do
  package_name 'python2-psycopg2' if node['platform_version'].to_i >= 8
end
```

File: test/cookbooks/app_test/recipes/app3.rb [hunk 6]
```diff
@@ -8,6 +8,7 @@
   end
 end
 
+node.override['percona']['version'] = '5.7'
```
Comment (ramereth) [inline test/cookbooks/app_test/recipes/app3.rb:11]: This shouldn't be needed anymore as you fixed this in osl-mysql for CentOS 8.

File: metadata.rb [hunk 7]
```diff
@@ -18,6 +19,7 @@
 depends          'osl-mysql'
 depends          'osl-nginx'
 depends          'osl-nodejs'
+depends          'percona'
```
Comment (ramereth) [inline metadata.rb:22]: You don't need to include this directly here as it will be pulled in via osl-mysql.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: recipes/default.rb (hunk 3 above)
Comment (TheREK3R) [reply recipes/default.rb:34]: There are a lot of missing packages on rhel8 I was hoping to ge  you to audit the packages I should include. I am not sure what is needed

Comment (ramereth) [reply recipes/default.rb:34]: Next time it would be ideal to have a comment here explaining that.

Comment (TheREK3R) [reply recipes/default.rb:34]: sorry

File: recipes/default.rb [hunk 8]
```diff
@@ -31,19 +34,31 @@
...
 ).each do |p|
   package p
 end
 
 # geo-django depends
-%w(gdal-python geos-python proj proj-nad postgresql-devel
-   freetype-devel libpng-devel libjpeg-turbo-devel postgis
-   python-psycopg2).each do |p|
+%w(geos-devel proj freetype-devel libpng-devel libjpeg-turbo-devel
+  ).each do |p|
```
[the remark this answers is not in the document]
Comment (TheREK3R) [reply recipes/default.rb:54]: Do you want this as a function to pich packages or to install them

[the remark this answers is not in the document]
Comment (TheREK3R) [reply recipes/default.rb:54]: eg package 7_packages 

or install_7_packages

[the remark this answers is not in the document]
Comment (ramereth) [reply recipes/default.rb:54]: it'll return an array of package names that you can then pass to the `package` resource. It'll be something like `osl_app_packages`

File: libraries/helpers.rb [hunk 9]
```diff
@@ -1,9 +1,45 @@
...
+        else
+          %w(
+            freetype-devel
+            libjpeg-turbo-devel
+            libpng-devel
+            postgresql-devel
+            proj
+            python3-gdal
+            python3-psycopg2
+          )
```
Comment (ramereth) [inline libraries/helpers.rb:37]: Please alphabetize.

File: recipes/app1.rb [hunk 10]
```diff
@@ -26,13 +26,13 @@
...
-  commands sudo_commands('openid-staging-unicorn', 'openid-staging-delayed-job')
+  commands osl_sudo_commands('openid-staging-unicorn', 'openid-staging-delayed-job')
   nopasswd true
 end
 
 sudo 'openid-production' do
   user 'openid-production'
-  commands sudo_commands('openid-production-unicorn',
+  commands osl_sudo_commands('openid-production-unicorn',
                          'openid-production-delayed-job')
```
Comment (ramereth) [inline recipes/app1.rb:36]: Let's just do it this way as it's cleaner.
```suggestion
  commands osl_sudo_commands('openid-production-unicorn', 'openid-production-delayed-job')
```

File: recipes/default.rb [hunk 11]
```diff
@@ -31,17 +34,26 @@
...
+  sqlite-devel
+  libyaml-devel
+  readline-devel
+  zlib-devel
+  libffi-devel
+  openssl-devel
+  automake
+  libtool
+  ImageMagick-devel
+  postgresql-devel
```
Comment (ramereth) [inline recipes/default.rb:47]: Alphabetize please.

File: recipes/default.rb [hunk 12]
```diff
@@ -31,17 +34,26 @@
...
 end
 
-# geo-django depends
-%w(gdal-python geos-python proj proj-nad postgresql-devel
-   freetype-devel libpng-devel libjpeg-turbo-devel postgis
-   python-psycopg2).each do |p|
-  package p
+package 'Dependency packages' do
+  package_name osl_app_packages()
+end
```
Comment (ramereth) [inline recipes/default.rb:54]:
```suggestion
package 'osl-app packages' do
  package_name osl_app_packages
end
```

File: test/integration/default/inspec/default_spec.rb [hunk 13]
```diff
@@ -1,9 +1,47 @@
...
+    describe package(p) do
+      it { should be_installed }
+    end
+  end
+else
+  %w(libpq-devel python2-psycopg2).each do |p|
+    describe package(p) do
+      it { should be_installed }
+    end
   end
```
Comment (ramereth) [inline test/integration/default/inspec/default_spec.rb:45]: Just move this into the case statement block above.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: libraries/helpers.rb (hunk 9 above)
Comment (TheREK3R) [reply libraries/helpers.rb:37]: Isnt it?

Comment (ramereth) [reply libraries/helpers.rb:37]: Oh apparently it already is :joy:

File: recipes/default.rb (hunk 11 above)
Comment (ramereth) [reply recipes/default.rb:47]: This still needs fixed.

Comment (TheREK3R) [reply recipes/default.rb]: sorry, thought this was in inspec

File: templates/default/app-nginx.erb [hunk 14]
```diff
@@ -1,4 +1,7 @@
+server_names_hash_bucket_size 128;
```
Comment (ramereth) [inline templates/default/app-nginx.erb:1]: Why is this needed?

Comment (TheREK3R) [reply templates/default/app-nginx.erb:1]: ill remove and see if the issue persists, but a while ago I was getting an error because that variable was too small, It might have been fixed upstream now though

Comment (TheREK3R) [reply templates/default/app-nginx.erb:1]: seems like this is still nesecary

**Review by ramereth — APPROVED** [review]
Please do a minor bump.

**Comment by TheREK3R** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v2.8.0.  Have a nice day!
